src/vs/workbench/api/browser/mainThreadEditorTabs.ts

```diff
@@ -42,13 +38,13 @@ export class MainThreadEditorTabs {
 		const tabs: IEditorTabDto[] = [];
 		for (const group of this._editorGroupsService.groups) {
 			for (const editor of group.editors) {
-				if (editor.isDisposed() || !editor.resource) {
+				if (editor.isDisposed()) {
 					continue;
 				}
 				tabs.push({
-					group: group.id,
-					name: editor.getTitle(Verbosity.SHORT) ?? '',
-					resource: EditorResourceAccessor.getOriginalUri(editor) ?? editor.resource,
+					viewColumn: editorGroupToColumn(this._editorGroupsService, group),
+					label: editor.getTitle(Verbosity.SHORT) ?? '',
+					resource: EditorResourceAccessor.getOriginalUri(editor),
```

Comment:
@bpasero just to double check that this is the URI that's also used for text documents?

Reply:
@jrieken yeah we might need to decide here or even have 2 properties: 
* `originalUri`: this MAY be the text document resource, it could also be the UI one used for the label if e.g. the user renamed a file
* `canonicalUri`: this is the text document resource

Reply:
I would suggest to go with `canonicalUri` than, at least only surface that in the UI for the sake of simplicity and to make it possible to associate a tab with a text document

Reply:
Yeah especially since `resource` for a text document is the canonical one it makes sense to me to use it here as well.